### DISCUSSION TITLE
Strange 404's in console

### DISCUSSION THREAD

**Discussion by ryanbahan** [vercel/next.js#14466] [Help]
Hey all,

I'm working on a little Pokedex project **[here](https://pokedex.ryanbahan.vercel.app/)**

And while my local version works fine, my prod build seems to log a ton of 404's that point to some strange .js files in chrome devtools Application > Frames > other. It looks like these js files are being generated from the fetched images, but I'm not sure how or why. 

The behavior is also inconsistent. Sometimes it logs `GET https://pokedex-7f1t60fsp.vercel.app/_next/static/Oz-Y64TTusNnDeJYbJMjb/pages/pokemon/001.js net::ERR_ABORTED 404` while other times it logs a series of `Failed to load resource: the server responded with a status of 404 ()`

Would love any insight/direction here. It's not affecting the app itself to my knowledge, but is driving me crazy.

**Comment by dkarkar92** [comment]
Are you using dynamic routing? I would check your `href` and `as` attributes on Link to make sure they're accurate. We had an issue where prod would result in several errors, turned out to be due to how the Next Link tag works with dynamic routing. 

https://nextjs.org/docs/api-reference/next/link#dynamic-routes

**Reply by ryanbahan** [reply]
@Enviii are you able to elaborate? I am using dynamic routing, and my route is currently set up as such: 

```
        <Link href={"/pokemon/" + num} as={"/pokemon/" + num}>
            <Section>
                <Num>{num}</Num>
                <img src={img} />
                <Name>{name}</Name>
                <P>{arrayToString(type)}</P>
            </Section>
        </Link>
```

I don't see anything in the docs to indicate I'm missing something.

**Comment by ryanbahan** [comment]
Update: changed the Link component as such:

```
        <Link href={"/pokemon?uri=" + num} as={"/pokemon/" + num}>
            <Section>
                <Num>{num}</Num>
                <img src={img} />
                <Name>{name}</Name>
                <P>{arrayToString(type)}</P>
            </Section>
        </Link>
```

And that removed 99% of the console errors. followed this guide: https://www.loudnoises.us/solving-next-static-pages-js-404-not-found/

However, there's still one lingering `Failed to load resource: the server responded with a status of 404 ()` now related to a pokemon.js file. I'm assuming this is b/c the route is /pokemon/[id], but I've updated all Link components and still get this error.

**Reply by grantjr1842** [reply]
Good looking pokedex!  If your single pokemon page is located in /pages/pokemon/[num].js then your Link should look like this

```
<Link href={"/pokemon/[num]"} as={"/pokemon/" + num}>
            <Section>
                <Num>{num}</Num>
                <img src={img} />
                <Name>{name}</Name>
                <P>{arrayToString(type)}</P>
            </Section>
        </Link>
```

EDIT:  I say this because I had a similar and decided to do a project search for "<Link" and went through every one to make sure they were following the correct format if it was a dynamic route.  Looks like you might have a <Link> component somewhere pointing to /pokemon.

Maybe double check that one.

**Reply by cooljameswei** [reply]
> Update: changed the Link component as such:
> 
> ```
>         <Link href={"/pokemon?uri=" + num} as={"/pokemon/" + num}>
>             <Section>
>                 <Num>{num}</Num>
>                 <img src={img} />
>                 <Name>{name}</Name>
>                 <P>{arrayToString(type)}</P>
>             </Section>
>         </Link>
> ```
> 
> And that removed 99% of the console errors. followed this guide: https://www.loudnoises.us/solving-next-static-pages-js-404-not-found/
> 
> However, there's still one lingering `Failed to load resource: the server responded with a status of 404 ()` now related to a pokemon.js file. I'm assuming this is b/c the route is /pokemon/[id], but I've updated all Link components and still get this error.

I met same error just now.
But it's working well while there's 404 error in console.
It's something like this..

Failed to load resource: the server responded with a status of 404 (Not Found)     http://localhost:3000/[object%20Object]

![image](https://user-images.githubusercontent.com/110060055/213088812-bb6803ea-f57e-41c3-b88e-2f773ecf45e5.png)
Looking forward to reply.
